eval/src/vespa/eval/tensor/dense/dense_tensor_address_combiner.h

```diff
@@ -18,57 +18,87 @@ namespace vespalib::tensor {
 class DenseTensorAddressCombiner
 {
 public:
-    using Address = DenseTensorCellsIterator::Address;
+    using Mapping = std::vector<std::pair<uint32_t, uint32_t>>;
 
 private:
-    enum class AddressOp { LHS, RHS, BOTH };
-
-    using CellsIterator = DenseTensorCellsIterator;
-
-    class AddressReader
-    {
-    private:
-        const Address &_address;
-        uint32_t       _idx;
-    public:
-        AddressReader(const Address &address) : _address(address), _idx(0) {}
-        Address::value_type nextLabel() { return _address[_idx++]; }
-    };
+    using Address = DenseTensorCellsIterator::Address;
+    using CellsRef = vespalib::ConstArrayRef<double>;
+    using size_type = eval::ValueType::Dimension::size_type;
 
-    std::vector<AddressOp> _ops;
+    const eval::ValueType &_rightType;
     Address                _combinedAddress;
+    CellsRef               _rightCells;
+    Address                _rightAddress;
+    std::vector<size_t>    _accumulatedSize;
```

Comment:
consider using 'right' in this name as well

Reply:
Changed in #4595